### ISSUE TITLE
 Issue with Global Constants as Default Values in Route Generation

### ISSUE THREAD

**Issue by srdrymn** [Milad-Akarie/auto_route_library#1918]
## Description

When using global constants as default values in constructors annotated with `@RoutePage()`, the generated route code does not include import statements for those constants, leading to unresolved references.

## Steps to Reproduce

1. Define a global constant in a separate file, for example, `constants.dart`:

```dart
const int appMaxWidth = 600;
```

1. Import this constant in a file where a page class is defined and annotate the page with @RoutePage(), using the global constant as a default value for a parameter:

```dart
import 'package:flutter/material.dart';
import 'package:auto_route/auto_route.dart';
import 'constants.dart'; // Import the file with the global constant

@RoutePage()
class MyPage extends StatefulWidget {
  const MyPage({
    Key? key,
    this.maxWidth = appMaxWidth, // Use the global constant as a default value
  }) : super(key: key);

  final int maxWidth;

  @override
  _MyPageState createState() => _MyPageState();
}
```

2. Run the code generation command for auto_route.

3. Observe that the generated route code for MyPage does not include an import statement for constants.dart, resulting in an unresolved reference to appMaxWidth.

## Expected Behavior
The generated route code should include all necessary import statements to resolve any references to global constants used as default values in route constructors.

## Actual Behavior
The generated route code lacks import statements for external files defining global constants, leading to unresolved references.

## Environment 
Flutter version: 3.19.5
Dart version: 3.3.3
auto_route version: 8.0.3
auto_route_generator version: 8.0.0

**Comment by Milad-Akarie** [comment]
Hey @srdrymn yes, I'm aware of this behavior, I should resolve the alias instead of the actual unit, will try to ship a fix with the next build.

**Comment by Milad-Akarie** [comment]
fixed in the latest auto_route version 9.0.0

**Comment by srdrymn** [comment]
@Milad-Akarie Could you please consider reopening this issue? I believe further investigation is needed.
